### ISSUE TITLE
Stubborn archive checksum errors

### ISSUE THREAD

**Issue by ghost** [rvm/rvm#2059]
I had the same problems with rvm a few months ago which caused me
to walk away, today I found myself trying again and hit the same issues.

``` bash
λ lsb_release -a
No LSB modules are available.
Distributor ID: LinuxMint
Description:    Linux Mint Xfce Edition
Release:    1
Codename:   debian

λ rvm version

rvm 1.21.14 (master) by Wayne E. Seguin <wayneeseguin@gmail.com>, Michal Papis <mpapis@gmail.com> [https://rvm.io/]

λ rvm --debug install 1.9.3
1.9.3 - install
Searching for binary rubies, this might take some time.
Remote file does not exist https://rvm.io/binaries/mint/1/x86_64/ruby-1.9.3-p448.tar.bz2
Remote file does not exist http://jruby.org.s3.amazonaws.com/downloads/ruby-1.9.3-p448.tar.bz2
Remote file does not exist http://binaries.rubini.us/mint/1/x86_64/ruby-1.9.3-p448.tar.bz2
rvm_remote_server_url3 not found
No remote file name found
No binary rubies available for: mint/1/x86_64/ruby-1.9.3-p448.
Continuing with compilation. Please read 'rvm help mount' to get more information on binary rubies.
__rvm_setup_compile_environment_setup ruby-1.9.3-p448
rvm_autolibs_flag=4
__rvm_setup_compile_environment_requirements ruby-1.9.3-p448
Installing requirements for mint, might require sudo password.
requirements code for mint loaded
Found required packages: bash, curl, patch, bzip2, ca-certificates, gawk, g++, gcc, make, libc6-dev, patch, openssl, ca-certificates, libreadline6, libreadline6-dev, curl, zlib1g, zlib1g-dev, libssl-dev, libyaml-dev, libsqlite3-dev, sqlite3, autoconf, libgdbm-dev, libncurses5-dev, automake, libtool, bison, pkg-config, libffi-dev.
Requirements installation successful.
__rvm_setup_compile_environment_movable ruby-1.9.3-p448
__rvm_setup_compile_environment_osx_compiler ruby-1.9.3-p448
__rvm_setup_compile_environment_architectures ruby-1.9.3-p448
rvm_architectures(0):.
__rvm_setup_compile_environment_gcc47plus ruby-1.9.3-p448
__rvm_setup_compile_environment_bison ruby-1.9.3-p448
__rvm_setup_compile_environment_smartos ruby-1.9.3-p448
__rvm_setup_compile_environment_sunos ruby-1.9.3-p448
__rvm_setup_compile_environment_openbsd ruby-1.9.3-p448
__rvm_setup_compile_environment_auto_conf_make ruby-1.9.3-p448
__rvm_setup_compile_environment_flags_docs ruby-1.9.3-p448
__rvm_setup_compile_environment_flags_tcltk ruby-1.9.3-p448
__rvm_setup_compile_environment_flags_shared_static ruby-1.9.3-p448
__rvm_setup_compile_environment_flags_threads ruby-1.9.3-p448
found compiler: /usr/lib/ccache/gcc
Installing Ruby from source to: /home/user1/.rvm/rubies/ruby-1.9.3-p448, this may take a while depending on your cpu(s)...
ruby-1.9.3-p448 - #downloading ruby-1.9.3-p448, this may take a while depending on your connection...
No archive, downloading
Running(11): curl --fail --location --max-redirs 10 --create-dirs -C - -o ruby-1.9.3-p448.tar.bz2 http://ftp.ruby-lang.org/pub/ruby/1.9/ruby-1.9.3-p448.tar.bz2
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 9816k  100 9816k    0     0   402k      0  0:00:24  0:00:24 --:--:--  338k
Downloaded archive checksum did not match, archive was removed!
If you wish to continue with not matching download add '--verify-downloads 2' after the command.

There has been an error fetching the ruby interpreter. Halting the installation.
__rvm_rm_rf already gone: /home/u
```

I've reinstalled rvm from the website, did an archive cleanup and reinstall and tried `rvm get head` but the checksum error persists.  
Using `--verify-downloads 2`  did allow the installation to complete, but the overall experience is "broken by default".

Glad to provide more info and help debug the problem, 
please bear in mind that I'm not proficient in ruby.

related(?) https://github.com/wayneeseguin/rvm/issues/1232

**Comment by revolter** [comment]
I have the same problem:

```
> rvm install 2.4 --debug
ruby-2.4.1 - install
ruby-2.4.1 - #already removed src/ruby-2.4.1
ruby-2.4.1 - #already removed rubies/ruby-2.4.1
Searching for binary rubies, this might take some time.
Running(16): curl --fail --location --max-redirs 10 --max-time 1800 --silent --insecure --location --list-only --max-time 5 --head https://rvm_io.global.ssl.fastly.net/binaries/osx/10.13/x86_64/ruby-2.4.1.tar.bz2 --insecure
Remote file does not exist https://rvm_io.global.ssl.fastly.net/binaries/osx/10.13/x86_64/ruby-2.4.1.tar.bz2
Running(16): curl --fail --location --max-redirs 10 --max-time 1800 --silent --insecure --location --list-only --max-time 5 --head https://s3.amazonaws.com/jruby.org/downloads/ruby-2.4.1.tar.bz2 --insecure
Remote file does not exist https://s3.amazonaws.com/jruby.org/downloads/ruby-2.4.1.tar.bz2
Travis binaries for OSX are not movable and can't be used outside of Travis environment. Skip that source.
No remote file name found
Running(16): curl --fail --location --max-redirs 10 --max-time 1800 --silent --insecure --location --list-only --max-time 5 --head https://rvm_io.global.ssl.fastly.net/binaries/osx/10.13/x86_64/ruby-2.4.1.tar.bz2 --insecure
Remote file does not exist https://rvm_io.global.ssl.fastly.net/binaries/osx/10.13/x86_64/ruby-2.4.1.tar.bz2
Running(16): curl --fail --location --max-redirs 10 --max-time 1800 --silent --insecure --location --list-only --max-time 5 --head https://s3.amazonaws.com/jruby.org/downloads/ruby-2.4.1.tar.bz2 --insecure
Remote file does not exist https://s3.amazonaws.com/jruby.org/downloads/ruby-2.4.1.tar.bz2
Travis binaries for OSX are not movable and can't be used outside of Travis environment. Skip that source.
No remote file name found
No binary rubies available for: osx/10.13/x86_64/ruby-2.4.1.
Continuing with compilation. Please read 'rvm help mount' to get more information on binary rubies.
Free disk space 16984MB, required 440MB.
__rvm_setup_compile_environment_setup ruby-2.4.1
rvm_autolibs_flag=4
__rvm_setup_compile_environment_movable_early ruby-2.4.1
__rvm_setup_compile_environment_system_early ruby-2.4.1
__rvm_setup_compile_environment_requirements ruby-2.4.1
Checking requirements for osx.
requirements code for osx loaded
requirements lib type set to osx_brew
brew seems to be writable
Found required packages: autoconf, automake, libtool, pkg-config, coreutils, libyaml, readline, libksba, openssl@1.1.
PATH=/usr/local/opt/coreutils/bin:/usr/local/opt/pkg-config/bin:/usr/local/opt/libtool/bin:/usr/local/opt/automake/bin:/usr/local/opt/autoconf/bin:/usr/local/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/X11/bin:/opt/local/bin:/opt/local/sbin:/bin:/usr/bin:/usr/sbin:/sbin:/Users/revolt/.local:/Users/revolt/.local/bin:/Users/revolt/Library/Python/2.7/bin:/usr/local/opt/qt/bin:/Users/revolt/.local/git-scripts:/Users/revolt/.gems/bin:/Users/revolt/.rvm/bin
rvm_configure_flags+=( --with-opt-dir=/usr/local/opt/libyaml )
rvm_configure_flags+=( --with-opt-dir=/usr/local/opt/readline )
rvm_configure_flags+=( --with-opt-dir=/usr/local/opt/libksba )
rvm_configure_flags+=( --with-opt-dir=/usr/local/opt/openssl@1.1 )
Certificates bundle '/usr/local/etc/openssl@1.1/cert.pem' is already up to date.
Requirements installation successful.
__rvm_setup_compile_environment_movable ruby-2.4.1
__rvm_setup_compile_environment_parse_name ruby-2.4.1
__rvm_setup_compile_environment_system ruby-2.4.1
__rvm_setup_compile_environment_architectures ruby-2.4.1
rvm_architectures(0):.
__rvm_setup_compile_environment_gcc47plus ruby-2.4.1
__rvm_setup_compile_environment_bison ruby-2.4.1
__rvm_setup_compile_environment_flags_docs ruby-2.4.1
__rvm_setup_compile_environment_flags_shared_static ruby-2.4.1
__rvm_setup_compile_environment_flags_threads ruby-2.4.1
__rvm_setup_compile_environment_compatibility_flag ruby-2.4.1
found compiler: gcc
Installing Ruby from source to: /Users/revolt/.rvm/rubies/ruby-2.4.1, this may take a while depending on your cpu(s)...
ruby-2.4.1 - #downloading ruby-2.4.1, this may take a while depending on your connection...
/Users/revolt/.rvm/scripts/functions/support: line 170: cd: /Users/revolt/.rvm/archives: No such file or directory
Load checksums for https://cache.ruby-lang.org/pub/ruby/2.4/ruby-2.4.1.tar.bz2
Fetch https://cache.ruby-lang.org/pub/ruby/2.4/ruby-2.4.1.tar.bz2 => ruby-2.4.1.tar.bz2
    ...no archive, downloading
Running(19): curl --fail --location --max-redirs 10 --max-time 1800 --connect-timeout 30 --retry-delay 2 --retry 3 --create-dirs -C - -o ruby-2.4.1.tar.bz2.part https://cache.ruby-lang.org/pub/ruby/2.4/ruby-2.4.1.tar.bz2
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 11.9M  100 11.9M    0     0  7706k      0  0:00:01  0:00:01 --:--:-- 7701k
Validating checksums for file /Users/revolt/.rvm/archives/ruby-2.4.1.tar.bz2
    ...checksums found in db
Validating md5 checksum
Calculate md5 checksum for /Users/revolt/.rvm/archives/ruby-2.4.1.tar.bz2
md5: /Users/revolt/.rvm/archives/ruby-2.4.1.tar.bz2: No such file or directory
    ...md5 checksum is not valid!!!
Downloaded archive checksum did not match, archive was removed!
If you wish to continue with not matching download add '--verify-downloads 2' after the command.

There has been an error fetching the ruby interpreter. Halting the installation.
__rvm_rm_rf already gone: /Users/revolt/.rvm/tmp/419*
```

``
> rvm --version
rvm 1.29.3 (latest) by Michal Papis, Piotr Kuczynski, Wayne E. Seguin [https://rvm.io]
```

**Comment by g4d** [comment]
I had same error.  Turns out it was due to brew not agreeing with a Mount Sierra upgrade.  Had to reinstall brew via this issue: https://gist.github.com/irazasyed/7732946#gistcomment-2298740 that also has more details.

**Comment by preethi-mohan2** [comment]
rvm install 2.7.1
Searching for binary rubies, this might take some time.
No binary rubies available for: osx/14.3/x86_64/ruby-2.7.1.
Continuing with compilation. Please read 'rvm help mount' to get more information on binary rubies.
Checking requirements for osx.
Certificates bundle '/usr/local/homebrew/etc/openssl@1.1/cert.pem' is already up to date.
Requirements installation successful.
Installing Ruby from source to: /Users/pmohan3/.rvm/rubies/ruby-2.7.1, this may take a while depending on your cpu(s)...
ruby-2.7.1 - #downloading ruby-2.7.1, this may take a while depending on your connection...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 14.0M  100 14.0M    0     0  9324k      0  0:00:01  0:00:01 --:--:-- 9324k
dyld[31329]: Library not loaded: /usr/local/homebrew/opt/openssl@3/lib/libcrypto.3.dylib
  Referenced from: <19825771-6618-331C-A043-548DAF97903B> /usr/local/Homebrew/Cellar/coreutils/9.5/bin/gsha512sum
  Reason: tried: '/usr/local/homebrew/opt/openssl@3/lib/libcrypto.3.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/usr/local/homebrew/opt/openssl@3/lib/libcrypto.3.dylib' (no such file), '/usr/local/homebrew/opt/openssl@3/lib/libcrypto.3.dylib' (no such file)
Downloaded archive checksum did not match, archive was removed!
If you wish to continue with not matching download add '--verify-downloads 2' after the command.

There has been an error fetching the ruby interpreter. Halting the installation.
